lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -1,49 +1,75 @@
 define((require, exports, module) ->
-    exports.setupPreviewer = (editor) ->
-        require("ace/ext/jquery")
-        katex = null
-
-        initKaTeX = (onLoaded) ->
-            # Adding KaTeX CSS
-            cssKatexPath = require.toUrl("../katex/katex.min.css")
-            linkKatex = $("<link>").attr("rel", "stylesheet").attr("href", cssKatexPath)
-            $("head").append(linkKatex)
-
-            # Adding CSS for demo formula
-            cssDemoPath = require.toUrl("./katex-demo.css")
-            linkDemo = $("<link>").attr("rel", "stylesheet").attr("href", cssDemoPath)
-            $("head").append(linkDemo)
-
-            # Adding DOM element to place formula into
-            span = $("<span>").attr("id", "formula")
-            $("body").append(span)
-
-            require(["ace/ext/katex/katex"], (katexInner) ->
-                katex = katexInner
-                onLoaded()
-                return
-            )
-            return
-
-        onLoaded = ->
-            selectedText = editor.getSelectedText()
-            try
-                katex.render(selectedText, $("#formula")[0])
-            catch e
-                $("#formula").text(e.message)
-            return
-
-        callback = (editor) ->
-            unless katex?
-                initKaTeX(onLoaded)
-                return
-            onLoaded()
-
-        editor.commands.addCommand({
-            name: "previewLaTeXFormula",
-            bindKey: {win: "Alt-p", mac: "Alt-p"},
-            exec: callback
-        })
+  exports.setupPreviewer = (editor) ->
+    katex = null
+
+    initKaTeX = (onLoaded) ->
+      # Adding CSS for demo formula
+      cssDemoPath = require.toUrl("./katex-demo.css")
+      linkDemo = $("<link>").attr(
+        rel: "stylesheet"
+        href: cssDemoPath
+      )
+      $("head").append(linkDemo)
+
+      # Adding DOM element to place formula into
+      a = $("<a>").attr(
+        href: "#"
+        id: "formula"
+        "data-toggle": "popover"
+      )
+      $("body").append(a)
+
+      require(["ace/ext/katex"], (katexInner) ->
+        katex = katexInner
+        onLoaded()
         return
+      )
+      return
+
+    options = {
```

Comment:
I suggest moving options object into `onLoaded` and replacing function in `content` with the result of explicit function call. This way you'll be able to catch katex error and show error message in the popover instead of logging it